ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteDialect.java

```diff
@@ -0,0 +1,533 @@
+			super( resultCursor, rowSelection );
+			this.queryReturns = queryReturns;
+		}
+		
+		@Override
+		TupleSnapshot createTupleSnapshot(List<?> value) {
+			Map<String, Object> map = new HashMap<>();
+			for (int i = 0; i < value.size(); i++) {
+				ScalarReturn ret = (ScalarReturn) queryReturns.get( i );
+				map.put( ret.getColumnAlias(), value.get( i ) );
+			}
+			return new MapTupleSnapshot(map);
+		}
+	}
+	
+	private class IgnitePortableResultCursor extends BaseResultCursor<Entry<String, BinaryObject>> {
+
+		public IgnitePortableResultCursor(QueryCursor<Entry<String, BinaryObject>> resultCursor, RowSelection rowSelection) {
+			super( resultCursor, rowSelection );
+		}
+		
+		@Override
+		TupleSnapshot createTupleSnapshot(Entry<String, BinaryObject> value) {
+			return new IgnitePortableTupleSnapshot( value.getValue() );
+		}
+	}
+	
+	private class IgnitePortableFromProjectionResultCursor extends BaseResultCursor<List<?>> {
+
+		public IgnitePortableFromProjectionResultCursor( QueryCursor<List<?>> resultCursor, RowSelection rowSelection ) {
+			super( resultCursor, rowSelection );
+		}
+
+		@Override
+		TupleSnapshot createTupleSnapshot( List<?> value ) {
+			return new IgnitePortableTupleSnapshot( value.get( 1 ) );
+		}
+	}
+	
+	public static class IgnitePortableTupleSnapshot implements TupleSnapshot {
+
+		private final BinaryObject binaryObject;
+		private final Set<String> columnNames;
+
+		public IgnitePortableTupleSnapshot(Object binaryObject) {
+			this.binaryObject = (BinaryObject) binaryObject;
+			if (binaryObject != null) {
+				this.columnNames = new HashSet<String>( this.binaryObject.type().fieldNames() );
+			}
+			else {
+				this.columnNames = Collections.emptySet();
+			}
+		}
+
+		@Override
+		public Object get(String column) {
+			return !isEmpty() ? binaryObject.field( column ) : null;
+		}
+
+		@Override
+		public boolean isEmpty() {
+			return binaryObject == null;
+		}
+
+		@Override
+		public Set<String> getColumnNames() {
+			return columnNames;
+		}
+	}
+
+	private class IgnitePortableAssociationSnapshot implements IgniteAssociationSnapshot<BinaryObject> {
+
+		private Map<RowKey, BinaryObject> portableMap = new HashMap<>();
+		
+		public IgnitePortableAssociationSnapshot(BinaryObject binaryObject, AssociationKey key) {
+			if (binaryObject != null) {
+				Map<BinaryObject, BinaryObject> associationMap = binaryObject.field( "ASSOCIATION" );
+				if (associationMap != null) {
+					for (BinaryObject portableKey : associationMap.keySet()) {
+						BinaryObject portableValue = associationMap.get( portableKey );
+						RowKey rowKey_tmp = portableKey.<RowKey>deserialize();
+						// sort arrays in RowKey for correct comparing of keys
+						String[] columnNames = rowKey_tmp.getColumnNames();
+						Object[] columnValues = rowKey_tmp.getColumnValues();
+						Arrays.sort( rowKey_tmp.getColumnNames() );
+						Arrays.sort( rowKey_tmp.getColumnValues() );
```

Comment:
This does not seem right, could you explain it a bit? 
I can see two problems here:
1) You changing the position of the values, this means that you don't know which value correspond to a column;
2) getColumnValues can contains different types, you cannot order them without having a classcast exception